crypto/kdf/hkdf.c

```diff
@@ -71,18 +71,19 @@ static int pkey_hkdf_ctrl(EVP_PKEY_CTX *ctx, int type, int p1, void *p2)
     HKDF_PKEY_CTX *kctx = ctx->data;
 
     switch (type) {
-    case EVP_PKEY_CTRL_HKDF_MD:
+    case EVP_PKEY_CTRL_KDF_MD:
+    case EVP_PKEY_CTRL_OLD_HKDF_MD:
         if (p2 == NULL)
             return 0;
 
         kctx->md = p2;
         return 1;
 
-    case EVP_PKEY_CTRL_HKDF_MODE:
+    case EVP_PKEY_CTRL_KDF_MODE:
```

Comment:
Is it inconceivable that some other KDF might have a mode setting as well?  If so, I don't see why we would need to make the tag specific to HKDF.